presto-main/src/main/java/com/facebook/presto/operator/ContinuousWork.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.operator.ContinuousWorkUtils.WorkState;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public interface ContinuousWork<T>
+{
+    /**
+     * Call the method to do the work.
+     * The caller can keep calling this method until it returns true.
+     * Result if ready when {@link ContinuousWork#process()} returns
```

Comment:
I cannot parse this sentence. Please rephrase.

edit: I get it. Result is ready means that you can obtain it using `getResult()` method. Please add that to javadoc.